App crashes when device is unplugged in saved experiments

**Actual Behaviour**

PSLab android application crashes when device is unplugged while an experiment is on going

**Expected Behaviour**

App should not crash and display a message instead

**Steps to reproduce it**

1. While performing an experiment, try unplugging the device

**LogCat for the issue**

```
08-20 01:39:41.383 14501-14501/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: org.fossasia.pslab, PID: 14501
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{org.fossasia.pslab/org.fossasia.pslab.activity.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.hardware.usb.UsbDeviceConnection.claimInterface(android.hardware.usb.UsbInterface, boolean)' on a null object reference
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2345)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2407)
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:149)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1324)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:211)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5321)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at java.lang.reflect.Method.invoke(Method.java:372)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1016)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
                                                    Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.hardware.usb.UsbDeviceConnection.claimInterface(android.hardware.usb.UsbInterface, boolean)' on a null object reference
                                                       at org.fossasia.pslab.communication.CommunicationHandler.open(CommunicationHandler.java:81)
                                                       at org.fossasia.pslab.communication.ScienceLab.<init>(ScienceLab.java:92)
                                                       at org.fossasia.pslab.others.ScienceLabCommon.openDevice(ScienceLabCommon.java:23)
                                                       at org.fossasia.pslab.activity.MainActivity.onCreate(MainActivity.java:107)
                                                       at android.app.Activity.performCreate(Activity.java:5933)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2407) 
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:149) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1324) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:211) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5321) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:372) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1016) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811) 
```

**Screenshots of the issue**

> Not available

**Would you like to work on the issue?**

Yes

--- a/app/src/main/java/org/fossasia/pslab/communication/PacketHandler.java
+++ b/app/src/main/java/org/fossasia/pslab/communication/PacketHandler.java
@@ -1,6 +1,7 @@
 package org.fossasia.pslab.communication;
 
 import android.util.Log;
+import android.widget.Toast;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
@@ -59,7 +60,7 @@ public class PacketHandler {
         if (!loadBurst) {
             try {
                 mCommunicationHandler.write(new byte[]{(byte) (val & 0xff)}, timeout);
-            } catch (IOException e) {
+            } catch (IOException | NullPointerException e) {
                 Log.e("Error in sending byte", e.toString());
                 e.printStackTrace();
             }
@@ -99,7 +100,7 @@ public class PacketHandler {
             try {
                 mCommunicationHandler.read(buffer, 1, timeout);
                 return buffer[0];
-            } catch (IOException e) {
+            } catch (IOException | NullPointerException e) {
                 e.printStackTrace();
                 return 3;
             }

--- a/app/src/main/java/org/fossasia/pslab/communication/PacketHandler.java
+++ b/app/src/main/java/org/fossasia/pslab/communication/PacketHandler.java
@@ -1,7 +1,6 @@
 package org.fossasia.pslab.communication;
 
 import android.util.Log;
-import android.widget.Toast;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;

--- a/app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java
+++ b/app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java
@@ -92,7 +92,7 @@ public class ScienceLab {
                 mCommunicationHandler.open();
                 //Thread.sleep(200);
                 mPacketHandler = new PacketHandler(50, mCommunicationHandler);
-            } catch (IOException e) {
+            } catch (IOException | NullPointerException e) {
                 e.printStackTrace();
             }
         }
@@ -1146,7 +1146,7 @@ public class ScienceLab {
             int vSum = mPacketHandler.getInt();
             mPacketHandler.getAcknowledgement();
             return vSum / 16.;
-        } catch (IOException e) {
+        } catch (IOException | NullPointerException e) {
             e.printStackTrace();
             Log.e(TAG, "Error in getRawAverageVoltage");
         }

